src/common/cryptography.js

```diff
@@ -32,6 +35,24 @@ module.exports.toBN = (value) => {
   return web3Utils.toBN(value);
 }
 
+
+/**
+ * Generate a hash of all inputs
+ * The Enigma contract uses the same logic to generate a matching taskId
+ *
+ * @param {array} inputsArray
+ * @return {string} hash of inputs
+ */
+module.exports.hashArray = (inputsArray) => {
+  let hexStr = '';
+  for (let e of inputsArray) {
+    e = utils.remove0x(e);
+    // since the inputs are in hex string, they are twice as long as their bytes
+    hexStr += (new BN(e.length/2).toString(16, 16)) + e;
```

Comment:
I don't like that it doesn't say anything about endianess, who decides that?

Reply:
We can be more explicit about this by doing something like:
```
hexStr += (new BN(e.length/2).toBuffer('be', 8).toString('hex')) + e;
```
(the 'be' means BigEndian)

but it looks uglier, i'll try to look more through the docs